public/app/plugins/panel/geomap/editor/StyleRuleEditor.tsx

```diff
@@ -41,25 +67,34 @@ export const StyleRuleEditor: FC<StandardEditorProps<FeatureStyleConfig, any, an
       onChange({
         ...value,
         check: {
-          ...value.check,
+          ...value.check!,
           operation: selection.value ?? ComparisonOperation.EQ,
-          property: value.check?.property ?? '',
-          value: value.check?.value ?? '',
         },
       });
     },
     [onChange, value]
   );
 
-  const onChangeComparisonValue = useCallback(
-    (e: ChangeEvent<HTMLInputElement>) => {
+  const onChangeValue = useCallback(
+    (selection?: SelectableValue) => {
+      onChange({
+        ...value,
+        check: {
+          ...value.check!,
+          value: selection?.value,
+        },
+      });
+    },
+    [onChange, value]
+  );
+
+  const onChangeNumericValue = useCallback(
+    (v?: number) => {
       onChange({
         ...value,
         check: {
-          ...value.check,
-          value: e.currentTarget.value,
-          operation: value.check?.operation ?? ComparisonOperation.EQ,
-          property: value.check?.property ?? '',
+          ...value.check!,
+          value: v!,
```

Comment:
Seems like we may be missing the `operation: selection.value ?? ComparisonOperation.EQ` which may fix the `==` issue